### PR TITLE
Wrapper fixes

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

**Comment by tbosch** [comment]
CI is red because of browserstack...

File: modules/@angular/compiler/src/metadata_resolver.ts [hunk 1]
```diff
@@ -751,6 +753,23 @@ function flattenArray(tree: any[], out: Array<any> = []): Array<any> {
   return out;
 }
 
+function dedupeArray(array: any[]): Array<any> {
+  const seen = new Set<any>();
```
Comment (vicb) [inline modules/@angular/compiler/src/metadata_resolver.ts]: `return Array.from(new Set(array))` ? 
we are building the set anyway

File: modules/@angular/compiler/src/metadata_resolver.ts [hunk 2]
```diff
@@ -413,14 +413,17 @@ export class CompileMetadataResolver {
       exportedModules: cpl.CompileNgModuleMetadata[]): cpl.TransitiveCompileNgModuleMetadata {
     // collect `providers` / `entryComponents` from all imported and all exported modules
     const transitiveModules = getTransitiveModules(importedModules.concat(exportedModules), true);
+    // attention: don't dedupe providers as we need to support overriding by adding the same
```
Comment (vicb) [inline modules/@angular/compiler/src/metadata_resolver.ts]: do we have a test ?

File: modules/@angular/compiler/src/metadata_resolver.ts [hunk 3]
```diff
@@ -751,6 +753,23 @@ function flattenArray(tree: any[], out: Array<any> = []): Array<any> {
...
+  const seen = new Set<any>();
+  return array.filter((entry) => {
+    const result = !seen.has(entry);
+    seen.add(entry);
+    return result;
+  });
+}
+
+
+function flattenAndDedupeArray(tree: any[], out: Array<any> = []): Array<any> {
```
Comment (vicb) [inline modules/@angular/compiler/src/metadata_resolver.ts]: Is the default used anywhere ?

File: modules/@angular/compiler/src/metadata_resolver.ts (hunk 1 above)
Comment (tbosch) [inline modules/@angular/compiler/src/metadata_resolver.ts]: changed

File: modules/@angular/compiler/src/metadata_resolver.ts (hunk 3 above)
Comment (tbosch) [inline modules/@angular/compiler/src/metadata_resolver.ts]: Removed

File: modules/@angular/compiler/src/metadata_resolver.ts (hunk 2 above)
Comment (tbosch) [inline modules/@angular/compiler/src/metadata_resolver.ts]: Reverted the change in this method

**Comment by vsavkin** [comment]
Merged https://github.com/angular/angular/commit/a178bc6c83276164ce69f7c9510eb9839eeb2f8d

**Pull request closed without merging**